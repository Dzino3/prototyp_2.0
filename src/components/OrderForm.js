import React, { useState} from 'react';
import './style/orderStyle.css';
import Sourced from './components-info/Sourced';
import Unsourced from './components-info/Unsourced';
import Axios from "axios"
import ListItem from './components-info/OrderListItem';


function OrderForm({user}) {
    const [detailsB, setDetailsB] = useState({order_numer: '', unit: ''});
    const [bestellInfo, setBestellInfo] = useState([]);

    const Checked = () =>{
        //console.log(user.unit);
        Axios.post("https://codux.herokuapp.com/order_number", {unit: user.unit, order_number: detailsB.order_numer, acc: user.acc}).then((response) =>{
            if (response.data.length !== 0){
                setBestellInfo(response.data[0]);
            }else{
                setBestellInfo([]);
            }
        })
    }


  return (
    <>
    <form className='orderForm'>
    <div className='orderForm-inner'>
        <div className='order-inner-inner'>
            <div className='orderForm-groupOut'>
                <h2>Search order</h2>
                <div className='orderForm-group'>
                    <label className='orderLabel'>Order number:</label>
                    <div className='orderForm-groupButton'>
                        <input className='orderInput' type="text" name='bestellnummer' id='bestellnummer' onChange={e => setDetailsB({...detailsB, order_numer: e.target.value})} value={detailsB.order_numer}></input>
                        <input className='checkButton' type="button" value="Check" onClick={Checked}></input>
                    </div>
                </div>
                
            </div>
            
        </div>
    </div>
    </form>
    <div className='orderItem'>{bestellInfo.length !== 0 ? <ListItem item={bestellInfo} selec={true}></ListItem> : <></>}</div>
    </>
  )
}

export default OrderForm
