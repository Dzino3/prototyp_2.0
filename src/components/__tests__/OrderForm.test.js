import React from "react";
import renderer from 'react-test-renderer';
import OrderForm from "../OrderForm.js";

describe(`OrderForm Rendering`, () => {

    it('should render OrderForm with user Hallo42', function () {
        let userX = {
            dealer: "Hallo42"
        };
        let tree = renderer.create(
            <OrderForm user={userX}></OrderForm>
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render OrderForm with empty user', function () {
        let userX = {}
        let tree = renderer.create(
            <OrderForm user={userX}></OrderForm>
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
})