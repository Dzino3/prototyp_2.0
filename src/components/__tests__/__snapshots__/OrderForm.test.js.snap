// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderForm Rendering should render OrderForm with empty user 1`] = `
[
  <form
    className="orderForm"
  >
    <div
      className="orderForm-inner"
    >
      <div
        className="order-inner-inner"
      >
        <div
          className="orderForm-groupOut"
        >
          <h2>
            Search order
          </h2>
          <div
            className="orderForm-group"
          >
            <label
              className="orderLabel"
            >
              Order number:
            </label>
            <div
              className="orderForm-groupButton"
            >
              <input
                className="orderInput"
                id="bestellnummer"
                name="bestellnummer"
                onChange={[Function]}
                type="text"
                value=""
              />
              <input
                className="checkButton"
                onClick={[Function]}
                type="button"
                value="Check"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>,
  <div
    className="orderItem"
  />,
]
`;

exports[`OrderForm Rendering should render OrderForm with user Hallo42 1`] = `
[
  <form
    className="orderForm"
  >
    <div
      className="orderForm-inner"
    >
      <div
        className="order-inner-inner"
      >
        <div
          className="orderForm-groupOut"
        >
          <h2>
            Search order
          </h2>
          <div
            className="orderForm-group"
          >
            <label
              className="orderLabel"
            >
              Order number:
            </label>
            <div
              className="orderForm-groupButton"
            >
              <input
                className="orderInput"
                id="bestellnummer"
                name="bestellnummer"
                onChange={[Function]}
                type="text"
                value=""
              />
              <input
                className="checkButton"
                onClick={[Function]}
                type="button"
                value="Check"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>,
  <div
    className="orderItem"
  />,
]
`;
